webpack/assets/javascripts/react_app/components/Editor/components/EditorNavbar.js

```diff
@@ -49,117 +52,143 @@ const EditorNavbar = ({
   isSelectOpen,
   showError,
   fetchAndPreview,
-}) => (
-  <div className="navbar navbar-form navbar-full-width navbar-editor">
-    <Nav className="nav nav-tabs nav-tabs-pf nav-tabs-pf-secondary">
-      <EditorRadioButton
-        stateView={selectedView}
-        btnView="input"
-        title={__('Editor')}
-        onClick={() => {
-          if (selectedView !== 'input') {
-            if (isRendering) toggleRenderView();
-            changeTab('input');
-          }
-        }}
-      />
-      <EditorRadioButton
-        stateView={selectedView}
-        disabled={!isDiff}
-        btnView="diff"
-        title={__('Changes')}
-        onClick={() => {
-          if (selectedView !== 'diff') {
-            changeTab('diff');
-          }
-        }}
-      />
-      {showPreview && (
-        <React.Fragment>
-          <EditorRadioButton
-            stateView={selectedView}
-            btnView="preview"
-            title={__('Preview')}
-            onClick={() => {
-              if (selectedView !== 'preview') {
-                if (!isRendering) toggleRenderView();
-                changeTab('preview');
-                if (selectedHost.id === '') fetchAndPreview(renderPath);
-              }
-            }}
-          />
-          {showHostSelector && (
-            <EditorHostSelect
-              show={selectedView === 'preview'}
-              open={isSelectOpen}
-              selectedItem={selectedHost}
-              placeholder={__('Select Host...')}
-              isLoading={isFetchingHosts}
-              onChange={host => previewTemplate({ host, renderPath })}
-              searchQuery={searchQuery}
-              onToggle={onHostSelectToggle}
-              onSearchChange={onHostSearch}
-              onSearchClear={onSearchClear}
-              options={isSearchingHosts ? filteredHosts : hosts}
-              key="hostsSelect"
+}) => {
+  const [safemode, setSafemode] = useState(isSafemodeEnabled);
+
+  const selectedRenderPath = useMemo(
+    () => (safemode ? safemodeRenderPath : renderPath),
+    [safemode, safemodeRenderPath, renderPath]
+  );
```

Comment:
Maybe we should directly set the `selectedRenderPath`?

e.g:
```suggestion
  const [selectedRenderPath, setRenderPath] = useState(renderPath);
```
and then on the checkbox call `setRenderPath`